app/src/org/commcare/activities/HomeScreenBaseActivity.java

```diff
@@ -194,10 +199,59 @@ private void loadInstanceState(Bundle savedInstanceState) {
     private void processFromExternalLaunch(Bundle savedInstanceState) {
         if (savedInstanceState == null && getIntent().hasExtra(DispatchActivity.WAS_EXTERNAL)) {
             wasExternal = true;
+            processSessionEndpoint();
             sessionNavigator.startNextSessionStep();
         }
     }
 
+    private void processSessionEndpoint() {
+        if (getIntent().hasExtra(SESSION_ENDPOINT_ID)) {
+            Endpoint endpoint = validateIntentForSessionEndpoint(getIntent());
+            if (endpoint != null) {
+                Vector<String> endpointArguments = endpoint.getArguments();
+                ArrayList<String> intentArguments = getIntent().getStringArrayListExtra(SESSION_ENDPOINT_ARGUMENTS);
+                HashMap<String, String> compositeArguments = new HashMap<>(endpointArguments.size());
+                for (int i = 0; i < endpointArguments.size(); i++) {
```

Comment:
I mentioned this in the `commcare-core` PR, but I think we should be setting the arguments by setting them as XPath variables on the evaluation context, since that will naturally and robustly substitute them in as needed as the stack ops are calculated.